### ISSUE TITLE
Extension for the page manager for handling error code

### ISSUE THREAD

**Issue by olape-git** [cmb69/pagemanager_xh#77]
Extension for the page manager to set the status code for specific pages (e.g. 403, 404, 410) without having to handle code snippets in the source code.

https://cmsimpleforum.com/viewtopic.php?f=16&t=18675&hilit=404&start=10#p84614

**Comment by cmb69** [comment]
So wie es jetzt ist, speichert der Pagemanager keine eigenen Daten, sondern ändert eben den Content und/oder die PageData. Das würde ich auch unbedingt so belassen. D.h. man müsste eine entsprechende Möglichkeit in page_params (oder einem anderen Plugin) ermöglichen. Ist es nicht in page_params, dann wäre ein entsprechendes User-Interface für den Pagemanager eben nur optional möglich (also nur wenn das entsprechende Plugin eben auch installiert ist).

Aber egal wo die Information gespeichert (und behandelt) wird, wird auf jeden Fall tricky das im Pagemanager zu unterstützen (bräuchte dann wohl ein zusätzliches Formular). Da wäre dann https://github.com/cmb69/pdeditor_xh vermutlich die bessere Variante (obwohl ich Zweifel habe, dass das Plugin wirklich nützlich ist).

Also warum nicht einfach ein zusätzliches Feld in page_params, dass dann im entsprechenden Pagedata-Tab geändert werden kann?

**Comment by olape-git** [comment]
> Also warum nicht einfach ein zusätzliches Feld in page_params, dass dann im entsprechenden Pagedata-Tab geändert werden kann?

Das war auch mein erster Gedanke.
Ich hätte es aber als besser empfunden, wenn man das in der Übersicht der Seiten erkennen könnte.
Aber vielleicht geht ja beides. In Pagedata aktivieren/bearbeiten und im Pagemanager anzeigen, wenn vorhanden.

Ich weiß, dass das vielleicht nicht so viel und oft genutzt wird. Aber im Forum kommt es immer mal wieder auf.
Und eine solche Möglichkeit wäre doch ein echter Vorteil für die User im Gegensatz zu Änderungen im Quelltext der Page.

**Comment by cmb69** [comment]
> und im Pagemanager anzeigen, wenn vorhanden.

Ja, wäre nett, ist aber wahrscheinlich sehr schwer umsetzbar, da jsTree eben keine weiteren Möglichkeiten bietet (der Pagemanager nutzt schon alles, was geht).

Eine ganz andere Möglichkeit, wie mir gerade einfällt, wäre https://github.com/cmb69/moved_xh; das ist eigentlich genau für solche Fälle gemacht (außer für 403).

**Comment by olape-git** [comment]
> Eine ganz andere Möglichkeit, wie mir gerade einfällt, wäre https://github.com/cmb69/moved_xh; das ist eigentlich genau für solche Fälle gemacht (außer für 403).

Das habe ich noch nie in der Hand gehabt. Dazu kann ich so nichts erst mal nichts sagen.
Allerdings würde ich es auch besser finden, wenn XH das OOTB könnte, was dann eher für page_params sprechen würde.
Aber ich seh es mir an.

**Update**
Also lokal, mit PHP 7.4.12 läuft es schon mal nicht. (Pluginstartseite und Konfiguration -->500)
Das muss ich mir in Ruhe ansehen.

-> Pfw_XH :-(

**Comment by olape-git** [comment]
So, ok. Soweit so gut. Aber.
Es ist ja Ziel, eigene Fehler-Seiten bereitzustellen. Das würde aber mit Moved_XH nur als Weiterleitung funktionieren und hätte damit automatisch 301 gefolgt von 200.
403, hast du selber schon vermerkt, geht nicht.
Und ich habe keine Möglichkeit, die XH-Fehler-Seiten aus der .htaccess zu nutzen, weil es diese einfach nicht gibt.
Das eigentlich angesprochene Problem ist also, Stand jetzt, auch mit Moved_XH noch unverändert vorhanden.

**Comment by olape-git** [comment]
Ich habe noch mal mit etwas Ruhe darüber nachgedacht.
Den Status für einzelne Seiten zu setzen, da ist sicher page_params die beste Stelle dafür.

In Moved_XH wäre es dann ein nettes Feature, wenn man statt der fest verdrahteten Meldung alternativ die entsprechenden selbst angelegten Pages auswählen könnte. Zumindest für 404 und 410. Für 410 wäre das ja theoretisch jetzt schon machbar, einfach als Weiterleitung, aber nicht wirklich schön.
Bei 403 wüsste ich jetzt auch keine Möglichkeit wie man das in Moved_XH umsetzen sollte.
Und für alles, was an XH vorbeigeht, muss man eh htaccess bemühen.

Im Pagemanager fände ich es immer noch gut, wenn irgendwie ersichtlich wäre, wenn ein abweichender Status gesetzt ist.
Und wenn es nur farblich ist.
